readers/mutation_source.hh

```diff
@@ -131,6 +146,55 @@ public:
         return fn(std::move(s), std::move(permit), range);
     }) {}
 
+    mutation_source(std::function<mutation_fragment_v1_stream(schema_ptr, reader_permit, partition_range, const query::partition_slice&, io_priority,
+                tracing::trace_state_ptr, streamed_mutation::forwarding)> fn)
+        : mutation_source([fn = std::move(fn)] (schema_ptr s,
+                    reader_permit permit,
+                    partition_range range,
+                    const query::partition_slice& slice,
+                    io_priority pc,
+                    tracing::trace_state_ptr tr,
+                    streamed_mutation::forwarding fwd,
+                    mutation_reader::forwarding) {
+        return fn(std::move(s), std::move(permit), range, slice, pc, std::move(tr), fwd);
+    }) {}
+    mutation_source(std::function<mutation_fragment_v1_stream(schema_ptr, reader_permit, partition_range, const query::partition_slice&, io_priority)> fn)
+        : mutation_source([fn = std::move(fn)] (schema_ptr s,
+                    reader_permit permit,
+                    partition_range range,
+                    const query::partition_slice& slice,
+                    io_priority pc,
+                    tracing::trace_state_ptr,
+                    streamed_mutation::forwarding fwd,
+                    mutation_reader::forwarding) {
+        assert(!fwd);
+        return fn(std::move(s), std::move(permit), range, slice, pc);
+    }) {}
+    mutation_source(std::function<mutation_fragment_v1_stream(schema_ptr, reader_permit, partition_range, const query::partition_slice&)> fn)
+        : mutation_source([fn = std::move(fn)] (schema_ptr s,
+                    reader_permit permit,
+                    partition_range range,
+                    const query::partition_slice& slice,
+                    io_priority,
+                    tracing::trace_state_ptr,
+                    streamed_mutation::forwarding fwd,
+                    mutation_reader::forwarding) {
+        assert(!fwd);
+        return fn(std::move(s), std::move(permit), range, slice);
+    }) {}
+    mutation_source(std::function<mutation_fragment_v1_stream(schema_ptr, reader_permit, partition_range range)> fn)
+        : mutation_source([fn = std::move(fn)] (schema_ptr s,
+                    reader_permit permit,
+                    partition_range range,
+                    const query::partition_slice&,
+                    io_priority,
+                    tracing::trace_state_ptr,
+                    streamed_mutation::forwarding fwd,
+                    mutation_reader::forwarding) {
+        assert(!fwd);
+        return fn(std::move(s), std::move(permit), range);
+    }) {}
```

Comment:
Do we really need this? We should migrate all users of the v1 `mutation_source::make_reader()` to the v2 equivalent and just have those few that need a v1 stream do the converting themselves.